### ISSUE TITLE
Problem with spacemacs \cup helm \cup agda2-mode

### ISSUE THREAD

**Issue by fredefox** [syl20bnr/spacemacs#10674]
#### Description :octocat:
When I'm using spacemacs with [agda-mode](https://github.com/agda/agda/) some helm commands works every other time. I'm suspecting this is related with the evil plugin, even though I have set spacemacs to operate with the emacs editing style.

#### Reproduction guide :beetle:
- Start Emacs
- `M-x agda2-mode`
- `C-x b` fine
- `C-x b` error. See stack trace below.

*Observed behaviour:* :eyes: :broken_heart:
I get the following error in the minibuffer: `Symbol's function definition is void: nil`.

*Expected behaviour:* :heart: :smile:
That I see the list of recent buffers.

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-function nil)
  nil()
  deactivate-input-method()
  activate-input-method(nil)
  evil-emacs-state(nil)
  evil-change-state(emacs nil)
  evil-change-to-initial-state(nil)
  evil-initialize-state()
  evil-local-mode(1)
  evil-initialize()
  evil-mode-enable-in-buffers()
  run-hooks(after-change-major-mode-hook)
  run-mode-hooks(helm-major-mode-hook)
  helm-major-mode()
  helm-create-helm-buffer()
  helm-initial-setup("/home/fredefox/cat/src/Cat/Category/Monad/Monoidal")
  helm-initialize(nil "/home/fredefox/cat/src/Cat/Category/Monad/" "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" helm-source-find-files)
  helm-internal(helm-source-find-files "/home/fredefox/cat/src/Cat/Category/Monad/" "Find files or url: " nil nil "*helm find files*" nil "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" nil)
  apply(helm-internal (helm-source-find-files "/home/fredefox/cat/src/Cat/Category/Monad/" "Find files or url: " nil nil "*helm find files*" nil "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" nil))
  helm(helm-source-find-files "/home/fredefox/cat/src/Cat/Category/Monad/" "Find files or url: " nil nil "*helm find files*" nil "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" nil)
  apply(helm (helm-source-find-files "/home/fredefox/cat/src/Cat/Category/Monad/" "Find files or url: " nil nil "*helm find files*" nil "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" nil))
  helm(:sources helm-source-find-files :input "/home/fredefox/cat/src/Cat/Category/Monad/" :case-fold-search smart :preselect nil :ff-transformer-show-only-basename t :default "/home/fredefox/cat/src/Cat/Category/Monad/Monoidal" :prompt "Find files or url: " :buffer "*helm find files*")
  helm-find-files-1("/home/fredefox/cat/src/Cat/Category/Monad/")
  (let* ((hist (and arg helm-ff-history (helm-find-files-history))) (default-input hist) (input (cond ((and (eq major-mode (quote dired-mode)) default-input) (file-name-directory default-input)) ((and (not (string= default-input "")) default-input)) (t (expand-file-name (helm-current-directory)))))) (set-text-properties 0 (length input) nil input) (helm-find-files-1 input))
  spacemacs/helm-find-files(nil)
  funcall-interactively(spacemacs/helm-find-files nil)
  call-interactively(spacemacs/helm-find-files nil nil)
  command-execute(spacemacs/helm-find-files)
```

I will report this in the agda-repo as well and see if they know something.

**Comment by Compro-Prasad** [comment]
Why do you want to press `C-x b` twice.

**Comment by fredefox** [comment]
I only want to press it once. But every other time i have to press it twice as per the description. `C-x b` was just an example. A minimal way to reproduce.

**Comment by syl20bnr** [comment]
Just noticed a weird thing in the report. Your `C-x b` is calling
`spacemacs/helm-find-files`, this is unexpected, `C-x b` should call
`helm-buffers-list`.

Do you have something in your configuration that could remap this key
binding ?

**Comment by fredefox** [comment]
Ah indeed. This is weird. Perhaps I mistakenly added a trace of doing `C-x f` in stead. I haven't remapped any keys though.

**Comment by fredefox** [comment]
You get a similar error with `visit-file` though.

**Comment by WhatisRT** [comment]
I have the same issue, with any helm command I tried. My `C-x b` calls `helm-buffers-list` though, but otherwise my backtrace looks very similar. As hinted by the backtrace, I also tried turning off the `Agda` input method, which seems to fix the issue, and I tried setting random input methods, which produce the same issue for me. So maybe this is caused by input methods in general?

Also, this affects only the second call to helm functions. So if I set the input method to japanese, I have to input `M-x` twice to observe the issue. It also affects different functions independently, so if I do a `C-x b` afterwards, it also works on the first time and fails the second time (or if I want to do `M-x set-input-method`, it works the first time, the next `M-x` fails right away, then `M-x set-input-method` fails, then `M-x` fails again, and then `M-x set-input-method` works again).
